packages/cli-kit/src/public/node/ruby.test.ts

```diff
@@ -181,4 +181,51 @@ describe('execCLI', () => {
     // Teardown
     process.env = originalEnv
   })
+
+  it('run bundled CLI2 when shopify user but not embedded env variable used', async () => {
+    // Setup
+    const originalEnv = process.env
+
+    // Given
+    const execSpy = vi.spyOn(system, 'exec')
+    process.env = {
+      ...originalEnv,
+      SHOPIFY_CLI_2_0_DIRECTORY: '/manual',
+      SHOPIFY_RUN_AS_USER: '0',
+      SHOPIFY_CLI_EMBEDDED_THEME_CLI: '0',
+    }
+
+    mockGemfileCreation()
+    vi.mocked(captureOutput).mockResolvedValueOnce('2.7.5')
+    vi.mocked(captureOutput).mockResolvedValueOnce('2.4.0')
+
+    // When
+    await execCLI2(['args'], {
+      token: 'token_0000_1111_2222_3333',
+      directory: './directory',
+    })
+
+    // Then
+    expect(execSpy).toHaveBeenLastCalledWith('bundle', ['exec', 'shopify', 'args'], {
+      stdio: 'inherit',
+      cwd: './directory',
+      env: {
+        ...process.env,
+        SHOPIFY_CLI_STOREFRONT_RENDERER_AUTH_TOKEN: undefined,
+        SHOPIFY_CLI_ADMIN_AUTH_TOKEN: undefined,
+        SHOPIFY_CLI_STORE: undefined,
+        SHOPIFY_CLI_AUTH_TOKEN: 'token_0000_1111_2222_3333',
+        SHOPIFY_CLI_RUN_AS_SUBPROCESS: 'true',
+        BUNDLE_GEMFILE: `/manual/Gemfile`,
+      },
+    })
+
+    // Teardown
+    process.env = originalEnv
+  })
 })
+
+function mockGemfileCreation() {
+  vi.mocked(file.fileExists).mockResolvedValue(true)
+  vi.mocked(file.readFile).mockResolvedValue('gemfile' as any)
+}
```

Comment:
I'd reconsider the testing strategy. While mocking dependencies of the subject under test is sometimes unavoidable, like it's the case of `system.exec`, which makes tests more unmaintainable because they depend on implementation details, there's no need to mock file IO APIs and we have plenty of examples throughout the codebase. We can create a temporary working directory that the test runs against. After the subject under test runs, you can go to that directory and check if the content of the `Gemfile` is the expected one.